test_istgt.sh

```diff
@@ -985,6 +985,141 @@ run_replica_connection_test ()
 	rm -rf ${replica1_vdev::-1}*
 }
 
+## wait_for_no_of_replicas will wait and return if specified no.of replias are
+## connected to it or else it returns error. It accepts replica_count as first
+## argument
+wait_for_no_of_replicas ()
+{
+	local replica_count=$1
+	local retry_cnt=5
+
+	for (( i = 0; i < $retry_cnt; i++ )) do
+		connected_replica_cnt=$(get_connected_replica_count)
+		if [ $connected_replica_cnt -eq $replica_count ]; then
+			break
+		fi
+		sleep 1
+	done
+	if [ $connected_replica_cnt -ne $replica_count ]; then
+		echo "Disconnected the replica: $replica3_id but still connected replica count is $connected_replica_cnt"
+		exit 1
+	fi
+}
+
+# run_resize_test will test the resize functionality when replicas are in different state
+run_resize_test ()
+{
+	local replica1_port="6161"
+	local replica2_port="6162"
+	local replica3_port="6163"
+	local replica1_id="6161"
+	local replica2_id="6162"
+	local replica3_id="6163"
+	local replica_ip="127.0.0.1"
+	local replica1_vdev="/tmp/test_vol1"
+	local replica2_vdev="/tmp/test_vol2"
+	local replica3_vdev="/tmp/test_vol3"
+	local CONF_FILE="/usr/local/etc/istgt/istgt.conf"
+	local current_size=""
+
+
+	DESIRED_REPLICATION_FACTOR=3
+	REPLICATION_FACTOR=3
+	CONSISTENCY_FACTOR=2
+	VOLUME_SIZE=1G
+
+	setup_test_env
+
+	local vol_name=$(cat $CONF_FILE | grep TargetName | awk '{print $2}')
+	local new_size=2
+	echo "Resize of volume when replicas are in different state"
```

Comment:
There won't be any replica connected at this stage.

Reply:
I will update the message